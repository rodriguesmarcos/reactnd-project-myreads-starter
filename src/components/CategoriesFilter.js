import React from 'react';
import PropTypes from 'prop-types';

const CategoriesFilter = ({categories, onChangeCategory, currCat}) => (
  <div>
    Filter by
    <select
      name="categoriesFilter"
      defaultValue={currCat}
      onChange={(e) => onChangeCategory(e.target.value)}
    >
      <option value="">All</option>
      {categories.map(c => (
        <option key={c.id} value={c.id}>{c.name}</option>
      ))}
    </select>
  </div>
);

CategoriesFilter.propTypes = {
  categories: PropTypes.array.isRequired,
  currCat: PropTypes.string.isRequired,
  onChangeCategory: PropTypes.func.isRequired,
}

export default CategoriesFilter;
