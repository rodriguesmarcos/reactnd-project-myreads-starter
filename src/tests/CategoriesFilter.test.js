import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CategoriesFilter from '../components/CategoriesFilter.js';
import books from './fixtures/books';
import shelves from './fixtures/shelves';


describe('<CategoriesFilter />', () => {

  const minProps = {
    categories: [{ id: 'computer', name: 'Computer' }],
    currCat: '',
    onChangeCategory: () => {},
  }

  test('renders without exploding', () => {
    const wrapper = shallow(<CategoriesFilter {...minProps}/>);
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });


  test('should call onChangeCategory', () => {
    const onChangeCategorySpy = jest.fn();
    const value = minProps.categories[0].id;
    const wrapper = shallow(<CategoriesFilter {...minProps} onChangeCategory={onChangeCategorySpy}/>);
    wrapper.find('select').simulate('change', {
      target: { value }
    });
    expect(onChangeCategorySpy).toHaveBeenCalledWith(value);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
