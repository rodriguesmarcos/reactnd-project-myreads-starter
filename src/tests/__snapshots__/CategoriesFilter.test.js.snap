exports[`<CategoriesFilter /> renders without exploding 1`] = `
<div>
  Filter by
  <select
    defaultValue=""
    name="categoriesFilter"
    onChange={[Function]}>
    <option
      value="">
      All
    </option>
    <option
      key="computer"
      value="computer">
      Computer
    </option>
  </select>
</div>
`;

exports[`<CategoriesFilter /> should call onChangeCategory 1`] = `
<div>
  Filter by
  <select
    defaultValue=""
    name="categoriesFilter"
    onChange={[Function]}>
    <option
      value="">
      All
    </option>
    <option
      key="computer"
      value="computer">
      Computer
    </option>
  </select>
</div>
`;
